### ISSUE TITLE
How to get the subtasks of a task throught the API?

### ISSUE THREAD

**Issue by thomaslombart** [Doist/todoist-api-typescript#148]
## Question description

**TL;DR: how can I get the subtasks of a given task in a performant way? I've been looking for a `subtaskIds` property or a `parentId` parameter when getting tasks but couldn't find any of these.**

---

My use case is for a Raycast extension allowing users to manage their Todoist tasks. One important feature missing is the support for sub-tasks. Here's the current behavior:

#### Todoist view

<img width="883" alt="CleanShot 2022-09-23 at 11 58 48@2x" src="https://user-images.githubusercontent.com/16003285/191937466-9fbe23d7-4c18-4b0d-a5b7-5cad95b6fb73.png">

#### Raycast view

<img width="1000" alt="todoist 2022-09-23 at 11 58 56" src="https://user-images.githubusercontent.com/16003285/191937497-79e4c2dc-edf4-4150-a4c7-46a80a15cade.png">

As you can see, the tasks in Raycast are all shown, including subtasks. While this is acceptable in this example, this can quickly become a problem as you add/nest subtasks.

Ideally, the user wouldn't see the subtasks at first. Instead, there would be an icon indicating how many subtasks a task has along with a command leading to a list view of all subtasks.

I've been looking for a `subtaskIds` property in the task scheme or a `parentId` parameter when getting tasks but couldn't find any of these.

The only solution I've found is to get all of the user's tasks and filter them by removing all the tasks that don't correspond to the `parentId` that's given. But this is not performant (especially if the user has hundreds of tasks).

**Comment by henningmu** [comment]
You can use `filter: '!subtask'` to avoid loading subtasks upfront. This doesn't give you any indication about whether a task has subtasks or how many, though :/ Also, this seems to be currently blocked by https://github.com/Doist/todoist-api-typescript/issues/149

How are you currently loading tasks? All at once, every time the extension is invoked? Could it make sense to use [the Sync API](https://developer.todoist.com/sync/v9/#sync) to load the items and then keep it up to date in the background somehow (e.g. with the [new Background Refresh API](https://developers.raycast.com/information/background-refresh))?

I think the main crux is the time spent on network, not the time spent iterating over the tasks. Maybe a loading indicator of sorts could also address this?

**Comment by thomaslombart** [comment]
The tasks are loaded every time the extension is invoked. I'm using filters or `getTasks` parameters depending on the command invoked. For example, I'm using `getTasks({ filter: "today|overdue" })` for the "Today Tasks" command, whereas for projects, I'm using the `projectId` parameter.

I could indeed use the Sync API since the API calls can be cached easily thanks to [Raycast's built-in hooks](https://developers.raycast.com/utilities/react-hooks/usecachedpromise). However, the Sync API isn't implemented in the TypeScript client, is it? If so, that means I'd lose the built-in types which would be a bummer.

That being said, I could use the background refresh API indeed. But I'm wondering a few things. Let's say I was to use the REST API to fetch all to-dos. Wouldn't it be bad for performance fetching all the user's to-dos? What about rate-limiting as well? 
Raycast's hooks will revalidate the data every time the hook is called, so at the initial render and when calling the `mutate` function (every time there's been an action such as updating the due date, the priority, etc.)
While it seems acceptable for a user who has few to-dos (let's say a hundred), I'm not sure this would be the same for a user that has thousands of to-dos.

**Comment by henningmu** [comment]
I think using the Sync API with [incremental sync](https://developer.todoist.com/sync/v9/#incremental-sync) is likely your best choice. It's similar to how the Todoist client apps operate: load (almost) all data upfront and then only load changes. 

> Sync API isn't implemented in the TypeScript client, is it?

Yes, that's indeed the main crux of the proposal :/ You'd have to implement this yourself. You can reuse some types with casting, but it won't be as seamless as this SDK.

> I could use the background refresh API indeed [...] Wouldn't it be bad for performance fetching all the user's to-dos?

Yes, loading all tasks every time will be a big hit on performance due to the network / API response times.

> What about rate-limiting as well?

Rate limits should be fine ([documentation](https://developer.todoist.com/sync/v9/#request-limits)), you can do 45 full syncs per 15 minutes or 450 partial syncs which should be more than enough. For the REST API, it'd be 450 requests per 15 minute interval ([docs](https://developer.todoist.com/rest/v2/#request-limits)).
